### ISSUE TITLE
AssemblyScript Public Meeting #21 - September 23rd, 2020

### ISSUE THREAD

**Issue by torch2424** [AssemblyScript/working-group#46]
# Date and Time

This public meeting will take place: September 23rd, 2020, 18:00 UTC (11:00 AM US PDT, UTC -8)

# General Agenda

* Agenda Items from comments left on this Github issue
* Additional in-meeting comments / discussion
* [If time allows] Recap of [WebAssembly CG meeting](https://github.com/WebAssembly/meetings) if anyone attended

Feel free to comment on this issue if you have any agenda items you would like to bring up. 
Meeting Notes will be placed on this issue for those who cannot make the meeting.

# Meeting Information

We will use the AssemblyScript discord for meetings: https://discord.gg/vSBbJ5T . We will meet in the "meeting" voice channel :+1: 

Anyone who is contributing to the AssemblyScript project, building something with AssemblyScript, interested in the Assembly project or WebAssembly in general, is welcome to join! 😄

Notes will be taken by the host, and posted after the meeting. Notes are free to be edited through comments on the meeting notes at a later time.

**Comment by torch2424** [comment]
Please notice! We changed from using Google Meet, to using Discord :smile:

**Comment by dcodeIO** [comment]
We'll eventually have to come up with a clear plan on when to enable complete / standardized features by default in the compiler. For instance, as seen on [webassembly.org's Roadmap](https://webassembly.org/roadmap/), just because a feature is standardized doesn't imply that it is broadly supported. Would love to hear your thoughts on this :)

**Comment by torch2424** [comment]
# Meeting Notes

* @dcodeIO : “We'll eventually have to come up with a clear plan on when to enable complete / standardized features by default in the compiler.”
   * For example, safari wont support wasm feature, should we support it? Feature detection?
   * The one in question is sign extension.
   * @torch2424 : We should wait for most platforms, unless someone opens an issue for their app
   * @dcodeIO : Well, less that, but more that if something is standardized, should we be pressuring standards to support them
   * Example that made sense to @torch2424 : 
      * Threads aren’t supported in all browsers, thus we have a flag. When is it ready to not have a flag?
   * @torch2424 : I can ask rust team what they do
   * @dcodeIO : cool, yeah let’s hear what they say
* @torch2424 : Wasm Live Talk
   * Went really well! People seemed to be very AS aware, which is great. 
   * Chat and panelists seem to really like my talk
   * Went very well!
* @DuncanUszkay1 : Closures
   * Following up in discussion in #development
   * TL;DR Reusing Scoped Locals are being a little bit funny
   * And GC is also a bit odd in the context of closures
   * @torch2424 : I think there might be value in getting something in the mainline for Closures, even if Closures isn’t highly optimized. As even the runtime itself is waiting on GC to be solid. And history kinda showed us there that building the MVP runtime was worth it.
   * @dcodeIO : Try compiling to the compiler to wasm :joy:
   * @dcodeIO : If someone can port a good GC to AS that would help.
      * Because locals are reference counted they are a bit complicated.
   * @torch2424 : Well, if we’re worried about optimizing Locals for runtime memory. How come?
   * @dcodeIO : IoT
   * @torch2424 : I kind of assume anyone doing IoT probably isn’t even using the runtime. That, and I think we shouldn’t optimize locals for IoT since I don’t think every new feature can support every Wasm use case optimally on day one (In the current state of wasm)
   * @dcodeIO : If we just had a fire and forget GC, that would be awesome.
   * @dcodeIO : Would be great if we could implement a better GC
      * Lmmix
      * ImmixGC or ImmixRC
      * Possibly better GC out there
   * @dcodeIO : Ideal Outcome, someone writes a polyfill for the Wasm GC Spec
      * Second outcome: someone writes a better GC in AS

**Comment by torch2424** [comment]
Closing this in favor of: https://github.com/AssemblyScript/working-group/issues/47

Feel free to continue discussions from this meeting here! :smile:
